airbyte-integrations/connectors/source-google-ads/source_google_ads/streams.py

```diff
@@ -472,3 +528,207 @@ class Labels(GoogleAdsStream):
+        self.parent_stream_name: str = self.parent_stream.name
+        self.parent_cursor_field: str = self.parent_stream.cursor_field
+        self.parent_sync_mode: SyncMode = SyncMode.incremental
+
+        super().__init__(**kwargs)
+        self.incremental_sieve_logger = cyclic_sieve(self.logger, 10)
+
+        self._state = {self.parent_stream_name: {customer.id: None for customer in self.customers}}
+
+    @property
+    @abstractmethod
+    def id_field(self) -> str:
+        "Name of field used for getting records by id"
+        pass
+
+    @property
+    @abstractmethod
+    def parent_id_field(self) -> str:
+        "Field name of id from parent record"
+        pass
+
+    @property
+    @abstractmethod
+    def resource_type(self) -> str:
+        "Resource type used for filtering parent records"
+        pass
+
+    @property
+    def state(self) -> MutableMapping[str, Any]:
+        return self._state
+
+    @state.setter
+    def state(self, value: MutableMapping[str, Any]):
+        self._state.update(value)
+        self.parent_stream.state = self._state.get(self.parent_stream_name, {})
+
+    def current_state(self, customer_id, default=None):
+        return self.parent_stream.current_state(customer_id, default)
+
+    def stream_slices(self, stream_state: Mapping[str, Any] = None, **kwargs) -> Iterable[Optional[MutableMapping[str, any]]]:
+        """
+        If state exists read updates from parent stream otherwise return slices with only customer id to sync all records for stream
+        """
+        if stream_state:
+            slices_generator = self.read_parent_stream(self.parent_sync_mode, self.parent_cursor_field, stream_state)
+            yield from slices_generator
+        else:
+            for customer in self.customers:
+                yield {"customer_id": customer.id, "updated_ids": set(), "deleted_ids": set(), "id_to_time": dict()}
+
+    def _process_parent_record(self, parent_record: MutableMapping[str, Any], child_slice: MutableMapping[str, Any]) -> bool:
+        """Process a single parent_record and update the child_slice."""
+        substream_id = parent_record.get(self.parent_id_field)
+        if not substream_id:
+            return False
+
+        # Save time of change
+        child_slice["id_to_time"][substream_id] = parent_record[self.parent_cursor_field]
+
+        # Add record id to list of changed or deleted items depending on status
+        slice_id_list = "deleted_ids" if parent_record.get("change_status.resource_status") == "REMOVED" else "updated_ids"
+        child_slice[slice_id_list].add(substream_id)
+
+        return True
+
+    def read_parent_stream(
+        self, sync_mode: SyncMode, cursor_field: Optional[str], stream_state: Mapping[str, Any]
+    ) -> Iterable[Mapping[str, Any]]:
+        for parent_slice in self.parent_stream.stream_slices(
+            sync_mode=sync_mode, cursor_field=cursor_field, stream_state=stream_state.get(self.parent_stream_name)
```

Comment:
is the sync mode ever not incremental? It was confusing to have to scroll up and down through multiple levels of indirection to see that it's only ever set to incremental, I think it would be easier to read if incremental was just hardcoded here

Reply:
Fixed